yt/frontends/stream/io.py

```diff
@@ -18,6 +18,8 @@ def __init__(self, ds):
     def _read_data_set(self, grid, field):
         # This is where we implement processor-locking
         tr = self.fields[grid.id][field]
+        if callable(tr):
```

Comment:
This, and the change below, is not the *most* elegant.  I'm not sure I have a better solution.

Reply:
it's typing madness, but I'd say that however inelegant, it's at least very compact and doesn't leak out of scope, so I think it's a reasonable approach overall

Reply:
OK